MailContainerTest/Services/MailTransferService.cs

```diff
@@ -1,93 +1,42 @@
 
-            switch (request.MailType)
+            var success = _requestApprovalService.TransferIsAllowed(sourceMailContainer, request);
+            if (success)
             {
-                case MailType.StandardLetter:
-                    if (mailContainer == null)
-                    {
-                        result.Success = false;
-                    }
-                    else if (!mailContainer.AllowedMailType.HasFlag(AllowedMailType.StandardLetter))
-                    {
-                        result.Success = false;
-                    }
-                    break;
+                sourceMailContainer.Capacity += request.NumberOfMailItems;
 
-                case MailType.LargeLetter:
-                    if (mailContainer == null)
-                    {
-                        result.Success = false;
-                    }
-                    else if (!mailContainer.AllowedMailType.HasFlag(AllowedMailType.LargeLetter))
-                    {
-                        result.Success = false;
-                    }
-                    else if (mailContainer.Capacity < request.NumberOfMailItems)
-                    {
-                        result.Success = false;
-                    }
-                    break;
+                _containerDataStore.UpdateMailContainer(sourceMailContainer);
 
-                case MailType.SmallParcel:
-                    if (mailContainer == null)
-                    {
-                        result.Success = false;
-                    }
-                    else if (!mailContainer.AllowedMailType.HasFlag(AllowedMailType.SmallParcel))
-                    {
-                        result.Success = false;
+                var destinationMailContainer = _containerDataStore.GetMailContainer(request.DestinationMailContainerNumber);
+                destinationMailContainer.Capacity -= request.NumberOfMailItems;
 
-                    }
-                    else if (mailContainer.Status != MailContainerStatus.Operational)
-                    {
-                        result.Success = false;
-                    }
-                    break;
+                _containerDataStore.UpdateMailContainer(destinationMailContainer);
             }
 
-            if (result.Success)
+            return new MakeMailTransferResult
             {
-                mailContainer.Capacity -= request.NumberOfMailItems;
-
-                if (dataStoreType == "Backup")
-                {
-                    var mailContainerDataStore = new BackupMailContainerDataStore();
-                    mailContainerDataStore.UpdateMailContainer(mailContainer);
-
-                }
-                else
-                {
-                    var mailContainerDataStore = new MailContainerDataStore();
-                    mailContainerDataStore.UpdateMailContainer(mailContainer);
-                }
-            }
-
-            return result;
+                Success = success,
```

Comment:
Here I would return more information, such as:
- a timestamp indicating when the request was processed
- a specific reason why the request failed, if so